sda-commons-server-kafka/src/test/java/org/sdase/commons/server/kafka/config/KafkaPropertiesTest.java

```diff
@@ -0,0 +1,133 @@
+package org.sdase.commons.server.kafka.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Properties;
+import org.junit.Test;
+import org.sdase.commons.server.kafka.KafkaConfiguration;
+import org.sdase.commons.server.kafka.KafkaProperties;
+
+public class KafkaPropertiesTest {
+  @Test
+  public void itShouldUseGlobalConfig() {
+    KafkaConfiguration config = new KafkaConfiguration();
+
+    config.getConfig().put("setting", "from.global");
+
+    KafkaProperties.forProducer(config).get("from.global");
+    KafkaProperties.forConsumer(config).get("from.global");
+    KafkaProperties.forAdminClient(config).get("from.global");
+  }
+
+  @Test
+  public void itShouldUseAdminConfigOverGlobalConfig() {
+    KafkaConfiguration config = new KafkaConfiguration();
+
+    AdminConfig adminConfig = new AdminConfig();
+    adminConfig.getConfig().put("setting", "from.admin");
+    config.setAdminConfig(adminConfig);
+
+    config.getConfig().put("setting", "from.global");
+
+    KafkaProperties.forProducer(config).get("from.global");
+    KafkaProperties.forConsumer(config).get("from.global");
+    KafkaProperties.forAdminClient(config).get("from.admin");
```

Comment:
To avoid the Sonar complaint:

```suggestion
    assertThat(KafkaProperties.forProducer(config).get("setting")).isEqualTo("from.global");
    assertThat(KafkaProperties.forConsumer(config).get("setting")).isEqualTo("from.global");
    assertThat(KafkaProperties.forAdminClient(config).get("setting")).isEqualTo("from.admin");
```